code/src/components/TodoInput.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from "react";
+import { useDispatch } from "react-redux";
+import { todos } from "../reducers/todos.js";
+
+
+export const TodoInput = () => {
+  const dispatch = useDispatch();
+  const [inputValue, setInputValue] = useState("");
+
+  // E står för Event
+  const handleOnSubmit = e => {
+    // Prevent page reload
+    e.preventDefault();
+    //console.log(inputValue)
+  
+
+  // This is where we will Dispatch the action to save new todo item
+    dispatch(
```

Comment:
nice work with comments to explain the code. But remove the console log 😉

Reply:
Also, you might want to check that the inputValue is not an empty string before saving the todo. Right now you can get "empty" todos in your list.
For example:
```
const handleOnSubmit = e => {
    e.preventDefault();
    
    if (!inputValue) return // this will just stop the handleOnSubmit function from going further

    dispatch(todos.actions.addTodo({
      description: inputValue,
      done: false,
    })
  );
  setInputValue("");
};
```